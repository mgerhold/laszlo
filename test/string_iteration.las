let text = "Hello, world!";
for i in 0..text.size {
    print(text[i]);
    print(" ");
}
println();

let i = 0;
for char in text {
    if i == 5 {
        text = "01234567890123456789";
    }
    if i == 10 {
        text = "ABC";
        println("for i == 10, char == " + char);
    }
    if i mod 2 == 0 {
        char = '!';
    }
    print(char + " ");
    i = i + 1;
}
println();
println(text);

for i in 0..26 {
    print('a' + i);
}
println();
