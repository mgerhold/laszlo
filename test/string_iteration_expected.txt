H e l l o ,   w o r l d ! 
! e ! l ! , ! 7 ! 9 for i == 10, char == 0
! 
ABC
abcdefghijklmnopqrstuvwxyz